cmake/linux.cmake

```diff
@@ -11,6 +11,11 @@ endif(CCACHE_FOUND)
 
 # set compiler flags
 IF(RTTR_ENABLE_OPTIMIZATIONS)
-	FORCE_ADD_FLAGS(CMAKE_C_FLAGS -ffast-math -mmmx -msse -mfpmath=sse -ggdb)
-	FORCE_ADD_FLAGS(CMAKE_CXX_FLAGS -ffast-math -mmmx -msse -mfpmath=sse -ggdb)
+	IF (PLATFORM_ARCH STREQUAL "armv7l")
+		FORCE_ADD_FLAGS(CMAKE_C_FLAGS -mcpu=cortex-a53 -mfloat-abi=hard -mfpu=neon-fp-armv8 -mneon-for-64bits -mtune=cortex-a53)
```

Comment:
This won't work for all armv7l systems. Besides that the current approach with the optimization flags is crappy as it assumes a given compiler.

Suggestion: Make an additional file `cmake/CompilerFlags.cmake` which you include from `CMakeLists.txt`. In that you add a cache-variable e.g. `TargetDevice` with a default value (maybe simply: "default") This should be used to specify for what RttR is compiled. You can then set the `STRINGS` property on that cache entry to specify possible values. Those should (for now) be `default RasPI3` (see https://blog.kitware.com/constraining-values-with-comboboxes-in-cmake-cmake-gui/ on how and why)

Later in that file you can use this variable to set your flags, when compiling for RasPI3. And while you are on it, you could also add appropriate flags for other RasPis. See https://gist.github.com/fm4dd/c663217935dc17f0fc73c9c81b0aa845 for those flags.

Reasoning: Most of those flags apply to a given compiler, not OS, so they should not be in OS specific files. I'll clean those up later. Having the ability to specify the target allows to pass default flags appropriate for that target.